src/Endatix.Api/Endpoints/Auth/Logout.cs

```diff
@@ -0,0 +1,45 @@
+using Endatix.Api.Infrastructure;
+using Endatix.Core.Infrastructure.Result;
+using Endatix.Core.UseCases.Identity.Login;
+using Endatix.Infrastructure.Identity.Authorization;
+using FastEndpoints;
+using MediatR;
+using Microsoft.AspNetCore.Http.HttpResults;
+
+namespace Endatix.Api.Endpoints.Auth;
+
+public class Logout(IMediator mediator) : EndpointWithoutRequest<Results<Ok<LogoutResponse>, BadRequest>>
+{
+    /// <summary>
+    /// Configures the endpoint settings for the logout functionality.
+    /// </summary>
+    public override void Configure()
+    {
+        Post("auth/logout");
+        Permissions(Allow.AllowAll);
+        Summary(s =>
+        {
+            s.Summary = "Logs out the authenticated user";
+            s.Description = "Initiates the logout process for the authenticated user.";
+            s.Responses[200] = "User logged out successfully.";
+            s.Responses[400] = "Invalid request or authentication state.";
+        });
+    }
+
+    /// <summary>
+    /// Executes the logout functionality
+    /// </summary>
+    /// <param name="cancellationToken">Cancellation token for the async operation.</param>
+    public override async Task<Results<Ok<LogoutResponse>, BadRequest>> ExecuteAsync(CancellationToken cancellationToken)
+    {
+
+        var logoutUserCmd = new LogoutCommand(User);
+        var logoutOperationResult = await mediator.Send(logoutUserCmd, cancellationToken);
+
+        var logoutResponse = logoutOperationResult.IsSuccess ?
```

Comment:
I know that we cannot directly use the logic of `ToEndpointResponse` as in the majority of our endpoints (e.g. for forms, definitions and submissions) but still I think we can move the logic with checking the status to the command handler.
This way we'll have the logic here cleaner on one side. Something more - currently `Result.Success` receives an object defined in the API assembly and `Result.Invalid` receives and object defined in the Core assembly. It will be good if we are more consistent here. Considering `LogoutResponse` has just a message, we can have a change in the core result to include an informative message for the for the happy path and construct the result object in the command handler with either the result with the message or the validation error.
This is just one proposal and although I haven't tried it, I think it shouldn't be hard. Additionally we can use the approach for other commands that don't return entities but need a massage.
What do you think?

Reply:
Yeah, I mutate the validation error by ignoring the original one, which is not very logical and now that I think about it , it's very shady business if done this way. Let me refactor to :
1. Propagate validation error from the original command
2. Encapsulate the message/no message flow within the extension method as you have suggested

Reply:
Cool, looks much better now! Thanks